tests/test_utils.py

```diff
@@ -0,0 +1,100 @@
+"""Test utils."""
+import io
+import os
+import json
+import pandas as pd
+import tempfile
+import unittest
+from bitarray import bitarray
+from clkhash.serialization import serialize_bitarray
+from clkhash.clk import generate_clk_from_csv
+from clkhash.schema import from_json_file
+from blocklib import generate_candidate_blocks
+from anonlinkclient.utils import deserialize_filters, generate_candidate_blocks_from_csv, combine_clks_blocks
+
+from tests import *
+
+
+class TestUtils(unittest.TestCase):
+
+    def test_deserialize_bitarray(self):
+        """Test deserialize bitarray."""
+        # create simple bitarray bloom filter
+        lst = [True, False, True, False] * 2
+        bf = bitarray(lst)
+        serialized_bf = serialize_bitarray(bf)
+        deserialized_bf = deserialize_filters([serialized_bf])
+        assert deserialized_bf == [bf]
+
+    def test_generate_candidate_blocks_from_csv_exception(self):
+        """Test exception case for generate candidate blocks from csv."""
+        data = [[1, 'Joyce'],
+                [2, 'Fred']]
+        input_f = io.StringIO(str(data))
+        schema_f = open(os.path.join(TESTDATA, 'bad-schema-v1.json'))
+
+        # test invalid header
+        header = 'Header'
+        with self.assertRaises(ValueError):
+            generate_candidate_blocks_from_csv(input_f, schema_f, header)
+
+    def test_combine_clks_blocks(self):
+        """Test combine clks and blocks."""
+        _, fname_clks = tempfile.mkstemp(suffix='.json', text=True)
+        _, fname_blks = tempfile.mkstemp(suffix='.json', text=True)
+        _, dummy = tempfile.mkstemp(suffix='.csv', text=True)
+        blocking_config = {
+            "type": "lambda-fold",
+            "version": 1,
+            "config": {
+                "blocking-features": [1, 2],
+                "Lambda": 2,
+                "bf-len": 1000,
+                "num-hash-funcs": 5,
+                "K": 5,
+                "input-clks": False,
+                "random_state": 0
+            }
+        }
+        filename = os.path.join(TESTDATA, 'dirty_1000_50_1.csv')
+        df = pd.read_csv(filename).set_index('rec_id')
+        data = df.to_dict(orient='split')['data']
+        blocking_obj = generate_candidate_blocks(data, blocking_config)
+        blocks = blocking_obj.blocks
+        for key in blocks:
+            blocks[key] = [int(x) for x in blocks[key]]
```

Comment:
Are we sure that block ids always be `int` as opposed to `str`?

Reply:
Here `x` is not block id but row index of original records. We go through all row indices that belongs to this block and convert them to integer if they are string